torch/csrc/autograd/python_engine.cpp

```diff
@@ -189,6 +190,15 @@ PyObject* THPEngine_run_backward(
                                              "allow_unreachable",
                                              "accumulate_grad",
                                              nullptr};
+
+  // This gets called while we are still in the main thread. The caveat of selecting
```

Comment:
I don't think we want this. This caller thread is the caller thread. We most likely don't want to change the name for that one.
Or if you want, you can do that on torch import: it will be the thread that did the `import torch`.

Reply:
@albanD thanks for taking a look. I think the thread that did the torch import is good IMO. Basically we want to capture the thread that launches the kernels for the forward pass and distinguish it from other threads as it's latency sensitive. Also we want to avoid having to instrument every application for the caller thread.

Reply:
Ok!
In that case I would move this call to https://github.com/pytorch/pytorch/blob/a3a81374843ba8c71e29bbcfd0aba0c11c71f644/torch/csrc/Module.cpp#L194

Reply:
I'm not sure however that if we do it on `import torch`, the name will be preserved when we spawn more processes for distributed training.

Reply:
@albanD thanks for the code pointer. I moved the name setting there and checked that it works as expected. I had to add the `null` byte to resizing the name string in `setThreadName`.